### PR TITLE
[CONTP-369] extract kubernetes resource parsers (pod, deployment and metadata) to a separate package

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

Extracts kubernetes resource parsers (pods, deployments and metadata) to their own package.

### Motivation

This PR doesn't change behaviour, but it is a preparatory step to allow other components to parse kubernetes resources and convert them to workloadmeta entities. 

The end goal is to be able to refactor some components (such as the orchestrator cluster checks) to compute resource tags on the fly instead of storing all tags for all resources in the tagger. 

As a concrete example, the cluster checks are usually interested only in unscheduled pods (pods that are not assigned to a node yet). However, by depending on the tagger, we will necessarily need to populate the tagger with tags for all pods in the cluster because there is no way for the tagger to distinguish between scheduled and unscheduled pods, which consumes a lot of memory, cpu and network.

The plan is to extract parsers to their own package, then expose some public utility methods in the tagger package that can help extract tags from a workloadmeta entity without having to populate tags in the tagger.

### Possible Drawbacks / Trade-offs

None

### Describe how to test/QA your changes

No need for qa, E2E tests are enough.

### CODE REVIEW

**Review by wdhif — APPROVED** [review]

File: comp/core/tagger/filters/filter_builder.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (davidor) [inline comp/core/tagger/filters/filter_builder.go:1]: Were this file and `filters.go` meant to be included in this PR? The title and the PR description don't seem to imply it.

**Review by davidor — APPROVED** [review]

**Pull request merged**